CONTRIBUTING.txt

```diff
@@ -0,0 +1,98 @@
+FreeCAD Contribution Process (FCP)
+==================================
+
+FreeCAD's contribution process is inspired by the Collective Code Construction Contract which itself is an evolution of the github.com Fork and Pull Model.
+
+0. Status
+
+FreeCAD is in a transition period. The following are to be regarded as GUIDELINES for contribution submission and acceptance. For historical reasons, the actual process may diverge from this process during the transition.  Such deviations should be noted and discussed whenever possible.
+
+1. Goals
+
+    The FreeCAD Contribution Process is expressed here with the following specific goals in mind:
+
+    1. To provide transparency and fairness in the contribution process.
+    2. To allow useful contributions to be included as quickly as possible.
+    3. To preserve and improve the code quality while encouraging appropriate experimentation and risk-taking
+    4. To minimize dependence on individual contributors by encouraging a large pool of active contributors
+    5. To be inclusive of many viewpoints and to harness a diverse set of skills
+    6. To provide an encouraging environment where contributors learn and improve their skills
+    7. To protect the free and open nature of the FreeCAD project
+
+2. Fundamentals
+
+    1. FreeCAD uses the git distributed revision control system.
+    2. Source code for the main application and related subprojects is hosted on github.com in the FreeCAD organization, hereafter referred to as 'the platform'.
+    3. Problems are discrete, well-defined limitations or bugs.
+    4. FreeCAD uses the platform's issue-tracking system to track problems
+    5. Patches are sets of code changes that resolve a single problem
+    6. FreeCAD uses the Pull Request workflow for evaluating and accepting patches
+
+
+3. Roles
+    1. "User": An member of the wider FreeCAD community who uses the software.
+    2. "Contributor":  A person who submits a patch that resolves a previously identified problem. Contributors do not have commit access to the repository unless they are also maintainers. Everyone, without distinction or discrimination, SHALL have an equal right to become a Contributor
+    3. "Maintainer": A person who merges patches. Maintainers may or may not be developers. Their role is to enforce the process. Maintainers have commit access to the repository.
+    4. "Administrator": Administrators have additional authority to maintain the list of designated maintainers.
+
+4. Licensing, Ownership, and Credit
+    1. FreeCAD uses the Lesser General Public License, version 2, or superior (LGPL2+) for all source code.  Additional details can be found in the LICENSE file.
```

Comment:
This document is mostly copied from a document with a GPLV3 license, if I understand it correctly.  Isn't it "derived" in the legal sense and thus must have the same or a compatible license (which LGPL would not be)?

Reply:
Agreed. There's still a lot of text here that is copied.  We should change it. Not just to avoid copying but to tailor the meaning to THIS project.  There is a lot about C4 that I like but I'm not comfortable adopting it verbatim. 
I don't know where the line is between derivation and re-implementation.

Reply:
In my experience the line is that the code/documentation is not "the same" when read by a "competent expert" in the field.  Re-writing all of it in our own words would probably be enough for a document, although a certain amount of "competent experts think alike" is permitted.  Copying word for word is derivation as best as I can understand the legal system :-)

Reply:
There are still a lot of C4 ideas here but the text has now been substantially changed.  I'm resolving this conversation.  Please note or re-open if you disagree.